node/resolvers/utils/quotes.ts

```diff
@@ -0,0 +1,135 @@
+export async function splitItemsBySeller({
+  ctx,
+  items,
+  quoteBySeller = {},
+  index = 0,
+}: {
+  ctx: Context
+  items: QuoteItem[]
+  quoteBySeller?: SellerQuoteMap
+  index?: number
+}): Promise<SellerQuoteMap> {
+  if (index >= items.length) return quoteBySeller
+
+  const item = items[index]
+  const { seller } = item
+
+  const next = async () =>
+    splitItemsBySeller({
+      ctx,
+      items,
+      quoteBySeller,
+      index: index + 1,
+    })
+
+  // The ternary check is to not request again from the same seller
+  const verifyResponse = quoteBySeller[seller]
```

Comment:
I think many calls would still be made if there was a seller who does not accept quotes with multiple items, since it would not be included in the `quoteBySeller` map.

Reply:
makes sense. let's analyze this.